### PR TITLE
abstract kubernetes clients

### PR DESCRIPTION
**Pull request by myaser**
teapot/issues#3254

### CODE REVIEW

File: pkg/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -33,3 +50,355 @@ func NewClient(host string, tokenSrc oauth2.TokenSource) (kubernetes.Interface,
...
+func (l Labels) String() string {
+	labels := make([]string, 0, len(l))
+	for key, val := range l {
+		labels = append(labels, fmt.Sprintf("%s=%s", key, val))
+	}
+	return strings.Join(labels, ",")
+}
+
+// Resource defines a minimal definition of a kubernetes Resource.
+type Resource struct {
```
Comment (myaser) [inline pkg/kubernetes/kubernetes.go:66]: moved from clusterpy

File: pkg/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -33,3 +50,355 @@ func NewClient(host string, tokenSrc oauth2.TokenSource) (kubernetes.Interface,
 func NewDynamicClient(host string, tokenSrc oauth2.TokenSource) (dynamic.Interface, error) {
 	return dynamic.NewForConfig(newConfig(host, tokenSrc))
 }
+
+type Labels map[string]string
```
Comment (myaser) [inline pkg/kubernetes/kubernetes.go:54]: moved from clusterpy

**Comment by linki** [comment]
👍

**Comment by myaser** [comment]
👍🏻

**Pull request merged**